pkg/proxy/iptables/proxier.go

```diff
@@ -948,6 +948,21 @@ func (proxier *Proxier) syncProxyRules() {
 	serviceNoLocalEndpointsTotalInternal := 0
 	serviceNoLocalEndpointsTotalExternal := 0
 
+	// If we are going to do stale chain cleanup, fetch the list of existing chains
+	// now. (Since "iptables-save" can take several seconds to run on hosts with lots
+	// of iptables rules, we don't bother to do this on every sync in large clusters.
+	// Stale chains will not be referenced by any active rules, so they're harmless
+	// other than taking up memory.)
```

Comment:
memory is an important factor at scale, there is also a existing work on reducing it just by removing fields from the objects on the informer storage https://github.com/kubernetes/kubernetes/pull/119556 and https://github.com/kubernetes/kubernetes/pull/118455

Reply:
ok, this comment is the same as below, I thought was something new

Reply:
what is the benefit of moving this here?

Reply:
Now we have access to `existingNATChains` while figuring out which rules we're going to create, so we can notice the case where we _thought_ a particular chain already existed but it turns out that it doesn't.

Reply:
> what is the benefit of moving this here?

(clarified in the commit message)